kernel/cmdline.cc

```diff
@@ -23,11 +26,61 @@ cmdlineread(mdev*, char *dst, u32 off, u32 n)
   return cc;
 }
 
+// Returns true if param is found in cmdline, false otherwise.
+// If found, writes the value of the first occurence to dst.
+// Expects cmdline to be a space-delimeted list of <param>=<value> pairs.
+static bool
+getvalue(const char* param, char* dst)
```

Comment:
Looking at this function more closely, I'm a bit concerned that it doesn't do bounds checks to ensure there aren't buffer overflows. If you look at [lib/string.c](https://github.com/fintelia/sv6/blob/master/lib/string.c) there are a few functions like `strncmp` and `safestrcpy` that will do checks for you